File: app/src/main/java/com/example/servicebookingandroid/Services/CommentService.java
package com.example.servicebookingandroid.Services;

import com.example.servicebookingandroid.Model.APIRequests.CommentRequest;
import com.example.servicebookingandroid.Model.CommentDto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface CommentService {

    @GET("/api/comment/get/{RequestId}}")
    Call<List<CommentDto>> getCommentsByRequestId(@Path("RequestId") String RequestId, @Header("Authorization") String Authorization);

    @GET("/api/comment/check/{RequestId}}")
    Call<CommentDto> checkDuplicateComment(@Path("RequestId") String RequestId, @Header("Authorization") String Authorization);

    @POST("/api/comment/post/{RequestId}}")
    Call<CommentDto> createComment(@Path("RequestId") String RequestId, @Header("Authorization") String Authorization, @Body CommentRequest commentRequest);
}

File: app/src/main/java/com/example/servicebookingandroid/Services/CommentService.kt
package com.example.servicebookingandroid.Services

import com.example.servicebookingandroid.Model.APIRequests.CommentRequest
import com.example.servicebookingandroid.Model.CommentDto
import retrofit2.Call
import retrofit2.http.*

interface CommentService {

    @GET("/api/comment/get/{RequestId}}")
    fun getCommentsByRequestId(@Path("RequestId") RequestId:String, @Header("Authorization") Authorization:String): Call<List<CommentDto>>

    @GET("/api/comment/check/{RequestId}}")
    fun checkDuplicateComment(@Path("RequestId") RequestId:String, @Header("Authorization") Authorization:String): Call<CommentDto>

    @POST("/api/comment/post/{RequestId}}")
    fun createComment(@Path("RequestId") RequestId:String, @Header("Authorization") Authorization:String, @Body commentRequest: CommentRequest): Call<CommentDto>
}